### PR TITLE
LPD-29607 Add invite users permission to publications 

### PR DESCRIPTION
**Pull request by brookedalton**
https://liferay.atlassian.net/browse/LPD-29607

### CODE REVIEW

**Comment by brookedalton** [comment]
https://github.com/liferay-publications/liferay-portal/assets/46908843/79c6b1f8-8313-4cf6-88ab-977fa06359ff

- Admin user can see all users in portal in dropdown
- Publication Admin can only see other publication users
- Publication Admin can also add non-publication users if their email is known
- Publication Viewer cannot add other publication users

File: modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/upgrade/v1_0_9/PublicationsAdminRoleUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			companyId -> {
+				List<Role> adminRoles = _roleLocalService.dslQuery(
+					DSLQueryFactoryUtil.select(
+						RoleTable.INSTANCE
+					).from(
+						RoleTable.INSTANCE
+					).where(
+						RoleTable.INSTANCE.name.eq(
+							PublicationRoleConstants.NAME_ADMIN)
+					));
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/upgrade/v1_0_9/PublicationsAdminRoleUpgradeProcess.java:50]: The name is a unique identifier for the roles, so there should be only one admin role per company. Use `roleLocalService.fetchRole(companyId, roleName)` to get the role.

**Review by pei-jung — CHANGES_REQUESTED** [review]

File: modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/action/ManageCollaboratorsMVCRenderCommand.java [hunk 2]
```diff
@@ -34,14 +39,28 @@ public String render(
 			RenderRequest renderRequest, RenderResponse renderResponse)
 		throws PortletException {
 
-		if (!CTPermission.contains(
-				PermissionThreadLocal.getPermissionChecker(),
```
Comment (brookedalton) [inline modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/action/ManageCollaboratorsMVCRenderCommand.java:38]: Earlier I was checking the INVITE_USERS permission with this CTPermission but it was incorrect and was checking `com.liferay.change.tracking#INVITE_USER` which yielded an error so I had to update the check to CTCollectionPermission

**Review by pei-jung — APPROVED** [review]
Nicely done, @brookedalton! 👍

**Comment by pei-jung** [comment]
ci:test:sf

**Comment by brookedalton** [comment]
ci:test:publications

**Comment by pei-jung** [comment]
ci:test:relevant

**Comment by brookedalton** [comment]
ci:stop

**Comment by brookedalton** [comment]
ci:test:relevant

**Comment by brookedalton** [comment]
ci:test:upgrade

**Comment by brookedalton** [comment]
UpgradeManagerTest seems to be a [flaky failure](https://testray.liferay.com/#/project/34754/cases/44364?filter=%7B%22dueStatus%22%3A%5B%22FAILED%22%5D%2C%22errors%22%3A%22testUpgradeManager%3A+java.lang.NullPointerException%22%7D&filterSchema=buildResultsHistory&page=1) 

[VirtualInstancesUpgrade#ViewVirtualInstancesArchive7310](https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/903831//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_VirtualInstancesUpgrade_ViewVirtualInstancesArchive7310_) failed due to a CI issue

**Comment by brookedalton** [comment]
ci:forward

**Pull request closed without merging**